models/experimental/vgg/demo/gs_demo.py

```diff
@@ -1,39 +1,39 @@
 # SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
-
 # SPDX-License-Identifier: Apache-2.0
 
 import torch
+import pytest
 
 from torchvision import models
 from loguru import logger
 
-import tt_lib
-
 from models.experimental.vgg.tt.vgg import *
-
-
-_batch_size = 1
-
-
-def test_gs_demo(device, imagenet_sample_input, imagenet_label_dict):
-    image = imagenet_sample_input
+from models.utility_functions import torch_to_tt_tensor, unpad_from_zero
+
+
+@pytest.mark.parametrize(
+    "batch_size",
+    (
+        (1),
+        (2),
+        (8),
+    ),
+)
+def test_gs_demo(device, imagenet_sample_input, imagenet_label_dict, batch_size):
+    images = batch_size * [imagenet_sample_input]
     class_labels = imagenet_label_dict
 
-    batch_size = _batch_size
     with torch.no_grad():
-        # TODO: enable conv on tt device after adding fast dtx transform
-        tt_vgg = vgg16(device, disable_conv_on_tt_device=True)
+        cache_path = "/mnt/MLPerf/tt_dnn-models/tt/VGG/vgg16/"
```

Comment:
Since external users do not have access to weka, datasets should be downloaded, converted to tt tensors (if needed) and cached when weka is not available when first time the model is run, and cache_path being set to the download path. 2nd run perf must match weka mounted 1st run. See falcon7b. It can also be a fixture.

Reply:
Updated the commit as suggested
Here is the commit link: [commit](https://github.com/tenstorrent-metal/tt-metal/compare/main...keerthana/vgg_move_intermediate_tensors_to_l1)